src/combobox/ResultItem.ts

```diff
@@ -0,0 +1,98 @@
+import { WidgetBase, onPropertiesChanged } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { DNode, PropertiesChangeEvent } from '@dojo/widget-core/interfaces';
+import { includes } from '@dojo/shim/array';
+
+import * as css from './styles/comboBox.css';
+
+/**
+ * @type ResultItemProperties
+ *
+ * Properties that can be set on a ResultItem component
+ *
+ * @property index			Position of this result in a list of results
+ * @property label			The label to display for this result item
+ * @property result			Data object associated with this item
+ * @property selected		Determines whether or not this item is selected
+ * @property skipResult		Called when a disabled result is selected to skip to next ResultMenuBase
+ * @property onMouseEnter	Called when mouse enters this result item
+ * @property onMouseDown	Called when mouse clicks this result item
+ * @property onMouseUp		Called when mouse releases this result item
+ */
+export interface ResultItemProperties extends ThemeableProperties {
+	index: number;
+	label: string;
+	result: any;
+	selected: boolean;
+	skipResult(): void;
+	onMouseEnter(index: number): void;
+	onMouseDown(event: MouseEvent): void;
+	onMouseUp(event: MouseEvent): void;
+};
+
+const ResultItemBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class ResultItem extends ResultItemBase<ResultItemProperties> {
+	@onPropertiesChanged
+	onPropertiesChanged(evt: PropertiesChangeEvent<this, ResultItemProperties>) {
+		if (includes(evt.changedPropertyKeys, 'selected')) {
+			const {
+				selected,
+				skipResult
+			} = this.properties;
+
+			selected === true && this.isDisabled() && skipResult();
+		}
+	}
+
+	isDisabled() {
+		return false;
```

Comment:
Consumers can override this. They'd create extend ResultItem and pass it into the ComboBox as a CustomResultItem. @matt-gadd and I chatted about this and favored this approach over passing a bunch of functions (like `isDisabled`) into the ComboBox properties directly.